website/content/maintainers/github/templates/required/contributing.md

```diff
@@ -0,0 +1,134 @@
+
+This is an optional section but we encourage you to think about your pull request process and set expectations for both contributors and reviewers.
+
+Instead of a fixed template, use the questions below as an exercise to uncover the unwritten rules and norms your project has for both reviewers and contributors. Using your answers, write a description of what a contributor can expect during their pull request.
+
+* When should contributors start to submit a PR - when it’s ready for review or as a work-in-progress?
+* How do contributors signal that a PR is ready for review or that it’s not complete and still a work-in-progress?
+* When should the contributor should expect initial review? The follow-up reviews?
+* When and how should the author ping/bump when the pull request is ready for further review or appears stalled?
+* How to handle stuck pull requests that you can’t seem to get reviewed?
+* How to handle follow-up issues and pull requests?
+* What kind of pull requests do you prefer: small scope, incremental value or feature complete?
+* What should contributors do if they no longer want to follow-through with the PR? Do maintainers sometimes commit to the PR directly to help get it merged? Or do maintainers close a PR if the contributor hasn’t responded in a specific time frame?
+* Once a PR is merged, what is the process for it getting into the next release?
+* When does a merged pull request end up in a release?
+
+## Development Environment Setup
+
+Provide enough information so that someone can find your project on the weekend and get set up, build the code, test it, and submit a pull request successfully without having to ask any questions. If there is a one-off tool they need to install, common error people run into, or useful script they should run, document that here. 
+
+Document any necessary tools, such as linters, or recommended IDE extensions. You don’t have to document the beginner’s guide to these tools, but how they are used within the scope of your project. This is a great place to include links to new user documentation videos and examples to get people started and understanding how to use the project
+
+You should explain how to do at least these basic tasks:
+
+* Get the source code
+* Retrieve any dependencies
+* Build the source code
+* Run the project locally
+* Test the source code, unit and "integration" or "end-to-end"
+* Generate and preview the documentation locally
+
+## Sign Your Commits
+
+Many new contributors do not have experience with having to sign or agree to a [Developer Certificate of Origin][DCO] (DCO) or Contributor License Agreement (CLA). Figuring out how to sign a commit can be a common stumbling block. Depending on which your project uses, keep either the text for the DCO or CLA sections.
+
+Since a DCO is pretty standardized, we have provided a template for projects that use them, but projects using a CLA should explain their process for signing a CLA as either an individual or a company.
```

Comment:
```suggestion
Since a DCO is pretty standardized, we have provided a template for projects that use them, but projects using a CLA should explain their process for signing a CLA as either an individual or a company.  Most CNCF projects will use [EasyCLA], which automates CLA signing via Github.
```